src/Microsoft.ML.HalLearners/HalLearnersCatalog.cs

```diff
@@ -31,15 +31,15 @@ public static class HalLearnersCatalog
         public static OlsLinearRegressionTrainer OrdinaryLeastSquares(this RegressionCatalog.RegressionTrainers catalog,
             string labelColumnName = DefaultColumnNames.Label,
             string featureColumnName = DefaultColumnNames.Features,
-            string weightColumn = null)
+            string exampleWeightColumnName = null)
         {
             Contracts.CheckValue(catalog, nameof(catalog));
             var env = CatalogUtils.GetEnvironment(catalog);
             var options = new OlsLinearRegressionTrainer.Options
             {
                 LabelColumn = labelColumnName,
                 FeatureColumn = featureColumnName,
-                WeightColumn = weightColumn != null ? Optional<string>.Explicit(weightColumn) : Optional<string>.Implicit(DefaultColumnNames.Weight)
+                WeightColumn = exampleWeightColumnName != null ? Optional<string>.Explicit(exampleWeightColumnName) : Optional<string>.Implicit(DefaultColumnNames.Weight)
```

Comment:
Uh oh. Am I wrong or is there *literally no way* to have non-weighted training if you happen to have a column named `Weight`????

It looks like there's an existing issue #1065 that is capturing this.

Reply:
Although I see it's incidentally assigned to you @abgoswam . 😄 Maybe after this PR we can think about how to address this.

Reply:
Sounds good. We should fix this behavior as part of  #1065   (in a separate PR).